### PR TITLE
Rename `whitelist_channels` to `allowlist_channels`

### PR DESCRIPTION
**Pull request by kenodegard**
### Description

Part of inclusive language efforts. This is a low-impact change as no other conda org project refers to `whitelist_channels`.

Ref: https://github.com/conda/conda/issues/11517

> **Warning**
> pre-commit/darker/black will not be satisfied until this has been reviewed. Prior to merge we will apply the necessary changes.

### Checklist - did you ...

- [x] Add a file to the `news` directory ([using the template](https://github.com/conda/conda/blob/main/news/TEMPLATE)) for the next release's release notes?
     
- [X] Add / update necessary tests?
- [X] Add / update outdated documentation?

### CODE REVIEW

File: conda/core/index.py [hunk 1]
```diff
@@ -27,14 +27,14 @@
 log = getLogger(__name__)
 
 
-def check_whitelist(channel_urls):
-    if context.whitelist_channels:
-        whitelist_channel_urls = tuple(concat(
-            Channel(c).base_urls for c in context.whitelist_channels
+def check_allowlist(channel_urls):
```
Comment (jaimergp) [inline conda/core/index.py:30]: I think mamba imports this function in their utils.py module:

https://github.com/mamba-org/mamba/blob/da44aadfaf88dfa3ce21656d671682626379cea6/mamba/mamba/utils.py#L71

We can add an alias for one release, with the corresponding deprecation warning?

Comment (kenodegard) [reply conda/core/index.py:30]: Oh good point

Comment (kenodegard) [reply conda/core/index.py:39]: Added

File: news/11647-allowlist_channels [hunk 2]
```diff
@@ -0,0 +1,19 @@
+### Enhancements
+
+* Rename `whitelist_channels` to `allowlist_channels`. (#11647)
```
Comment (beeankha) [inline news/11647-allowlist_channels:3]: I might reword this to say that we're changing any instances of `whitelist` to `allowlist` and `blacklist` to `denylist`

**Review by beeankha — APPROVED** [review]
Left a non-blocking suggestion for the news file; otherwise LGTM!

File: news/11647-allowlist_channels (hunk 2 above)
Comment (kenodegard) [reply news/11647-allowlist_channels:3]: Hehe, I explicitly wrote this with the opposite mindset; outline the actual change, not the policy/objective that resulted in the change. I wrote it this way since the change only resulted in one meaningful variable change. 🤷🏻 

I have no strong opinion here so if others agree that this should be reworded I'll make the change.

Comment (jezdez) [reply news/11647-allowlist_channels:3]: I think this is fine as written, it's not worth adding troll traps into the changelog. The issues and pull requests have more details than whatever we reasonably would have to write in the changelog to explain the policy change about naming things.

**Review by jezdez — APPROVED** [review]
🚀
